bundles/org.openhab.binding.venstarthermostat/src/main/java/org/openhab/binding/venstarthermostat/internal/handler/VenstarThermostatHandler.java

```diff
@@ -318,23 +336,93 @@ private VenstarSystemMode getSystemMode() {
         return infoData.getMode();
     }
 
-    private void updateThermostat(int heat, int cool, VenstarSystemMode mode) {
+    private VenstarAwayMode getAwayMode() {
+        return infoData.getAway();
+    }
+
+    private void updateSettings(VenstarAwayMode away) {
+        // this function corresponds to the thermostat local API POST /settings instruction
+        // the function can be expanded with other parameters which are changed via POST /settings
+        Map<String, String> params = new HashMap<>();
+        params.put("away", "" + away.mode());
+        String updateURLextension = "settings";
+        updateThermostat(updateURLextension, params);
+    }
+
+    private void updateControls(int heat, int cool, VenstarSystemMode mode) {
+        // this function corresponds to the thermostat local API POST /control instruction
+        // the function can be expanded with other parameters which are changed via POST /control
         Map<String, String> params = new HashMap<>();
-        log.debug("Updating thermostat {}  heat:{} cool {} mode: {}", getThing().getLabel(), heat, cool, mode);
         if (heat > 0) {
             params.put("heattemp", String.valueOf(heat));
         }
         if (cool > 0) {
             params.put("cooltemp", String.valueOf(cool));
         }
         params.put("mode", "" + mode.mode());
+        String updateURLextension = "control";
+        updateThermostat(updateURLextension, params);
+    }
+
+    private void updateThermostat(String updateType, Map<String, String> params) {
+        // this function is called from either updateControl or updateSettings and sends the command to the thermostat
+        // add other API variables here if they get programmed in future
+
+        // settings
+        VenstarAwayMode away = infoData.getAway();
+
+        // controls
+        double heat = infoData.getHeattemp();
+        double cool = infoData.getCooltemp();
+        VenstarSystemMode mode = infoData.getMode();
+
+        if (updateType == "settings") {
```

Comment:
So all the logic for settings should be scoped to the `updateSettings` method , no need to have it spread out in 2 functions.